server/src/main/java/io/crate/types/DoubleType.java

```diff
@@ -160,8 +159,8 @@ public Double implicitCast(Object value) throws IllegalArgumentException, ClassC
         } else if (value instanceof String s) {
             return JavaDoubleParser.parseDouble(s);
         } else if (value instanceof BigDecimal bigDecimalValue) {
-            if (DOUBLE_MAX.compareTo(bigDecimalValue.toBigInteger()) <= 0
-                || DOUBLE_MIN.compareTo(bigDecimalValue.toBigInteger()) >= 0) {
+            if (DOUBLE_MAX.compareTo(bigDecimalValue) <= 0
+                || DOUBLE_MIN.compareTo(bigDecimalValue) >= 0) {
                 throw new IllegalArgumentException(getName() + " value out of range: " + value);
```

Comment:
CompareTo must be prefeered over equals as it makes 2.0 and 2.00 same.

See https://github.com/eobermuhlner/big-math/issues/67 and compareTo javadocs

`Two BigDecimal objects that are equal in value but have a different scale (like 2.0 and 2.00) are considered equal by this method. Such values are in the same cohort. `

Reply:
Maybe worth having this as a comment in the code?

Reply:
Sounds good, added